### ISSUE TITLE
Support $ENV var for postgres connection string in config file

### ISSUE THREAD

**Issue by ARolek** [terranodo/tegola#95]
A recent conversation about using tegola with docker brought up the issue of passing config files to containers. It's a terrible idea to build a container with a config file that includes database connection string parameters, but containers can easily accept environment variables. Additionally it would be nice to be able to share config files for complex data configuration (i.e. OSM data). So the idea to support `$ENV` variables in the config file was proposed.

Initially this would be a modification to the connection string for the Postgres data provider. Currently we break out all the connection variables into:

- host
- port
- database
- user
- password

This would be converted to the Postgres documented [connection string format](https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING):

```
postgresql://[user[:password]@][netloc][:port][/dbname][?param1=value1&...]
```

A PostGIS data provider config would then look:

```toml
[[providers]]
name = "test_postgis"   # provider name is referenced from map layers
type = "postgis"        # the type of data provider.
connStr = "tegola:password@localhost:5432/tegola"
srid = 3857             # The default srid for this provider. If not provided it will be WebMercator (3857)
```

And using the environment variable:

```toml
[[providers]]
name = "test_postgis"   # provider name is referenced from map layers
type = "postgis"        # the type of data provider.
connStr = "$PG_CONN_STRING"
srid = 3857             # The default srid for this provider. If not provided it will be WebMercator (3857)
```
Our config parser would need to sniff for strings prefixed with "$" and then fill the value with the value set in the environment variable.

Alternatively, we could keep the config as is and allow the user to pass each parameter in via ENV variables.

**Comment by mojodna** [comment]
In the past, I've used [`pgexplode`](https://github.com/stamen/pgexplode) and [`interp`](https://github.com/stamen/interp) (which includes `pgexplode`'s functionality) to let me use `DATABASE_URL` in my environment and have it interpolated into config files w/o native environment support by treating them as handlebars templates.

**Comment by JivanAmara** [comment]
See https://github.com/terranodo/tegola/pull/237, this change allows arbitrary environment variables to inject content into the config file.

**Comment by ARolek** [comment]
Addressed in #237 which is now in the v0.6.0 branch.
